nflow-engine/src/main/java/io/nflow/engine/NflowEngine.java

```diff
@@ -0,0 +1,109 @@
+package io.nflow.engine;
+
+import io.nflow.engine.config.EngineConfiguration;
+import io.nflow.engine.config.NFlow;
+import io.nflow.engine.internal.executor.WorkflowDispatcher;
+import io.nflow.engine.internal.storage.db.SQLVariants;
+import io.nflow.engine.service.*;
+import io.nflow.engine.workflow.definition.AbstractWorkflowDefinition;
+import io.nflow.engine.workflow.definition.WorkflowState;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import javax.sql.DataSource;
+import java.util.Collection;
+
+/**
+ * NflowEngine starts up nflow-engine with given database and workflow definitions.
+ *
+ * Use this if you want to start just the nflow-engine.
+ */
+public class NflowEngine implements Runnable {
```

Comment:
I would add `AutoClosable` interface so that if one uses the NflowEngine in some tests or other places the code to handle shutdown will be much cleaner.

Reply:
Implemented